### ISSUE TITLE
fish completions appearing in Node.js interactive shell

### ISSUE THREAD

**Issue by probablykasper** [fish-shell/fish-shell#7593]
This bug happens in iTerm2 and VSCode's integrated terminal, but not in Terminal.app for whatever reason.

Recording: https://asciinema.org/a/tNyTa0ZELB4qVN7VfwD8WYfLE
In this recording, the completions are from [fnm]([fnm](https://github.com/Schniz/fnm)). I've also encountered this with [Docker's](https://docs.docker.com/compose/completion/) and [Poetry's completions](https://python-poetry.org/docs/#enable-tab-completion-for-bash-fish-or-zsh).

### Reproduction:
Alternatively, you can install [fnm]([fnm](https://github.com/Schniz/fnm))'s completions and follow the recording.
1. Install [Node.js](https://nodejs.org)
2. Install completions
3. Run `node` to enter Node.js' interactive shell
4. Type something to trigger completions

### Versions
- macOS 10.14.6 x64
- fish 3.1.2
- `echo $TERM` gives me `xterm-256color` in 3 terminals

**Comment by faho** [comment]
>In this recording, the completions are from fnm. 

What is `fnm`?

I can't reproduce this with fish 3.1.2 or from master. It looks like fish isn't properly giving up the terminal. In that case it would be important to know what `node` is - what does `type node` say?

**Comment by probablykasper** [comment]
@faho `fnm` is a version manager for Node.js (perhaps a bad call to pick something related to Node.js, but as mentioned Docker's and Poetry's completions cause the bug as well)

Could be that you're not typing something that triggers it. It's not too obvious what to type to trigger it.

`node` is a binary for running Node.js, just like `python` for instance. This is what I get from running `type node`:
```
node is /var/folders/44/3sxbyc2d6wg90mj1s7s59bnm0000gn/T/fnm_multishell_49720_1609430092718/bin/node
```

**Comment by faho** [comment]
>node is a binary for running Node.js, just like python for instance

Oh, I know what node is, the question was if it's a wrapper function or something instead of a binary directly.

Apparently it's a binary.

>Could be that you're not typing something that triggers it. It's not too obvious what to type to trigger it.

This looks like both node and fish receive your input, so to trigger it I would expect anything that prints a suggestion in fish would do - so if you e.g. typed `echo ` it should show up (as long as you've used `echo` in fish previously).

-----------

Also how are you starting fish? Is it your login shell, or do you start it via .bashrc or as yourr terminal's command or similar or manually?

**Comment by probablykasper** [comment]
@faho It might not be a `fish` issue afterall. This bug keeps perplexing me because it's hard to pinpoint it, I've 180'd so many times on what I think causes it.

Earlier when I tried, for some reason I wasn't able to reproduce this in `bash`, in `fish` on Terminal.app and when starting fish using `sh -c 'env HOME=$(mktemp -d) fish'`, which made me believe it had to do with `fish`. I was able to now though, so that rules out `fish`. I'm pretty sure the issue is just with `node` itself.

Sorry for that mistake, but thanks for looking into it regardless!
